web/app/js/components/Navigation.jsx

```diff
@@ -260,62 +359,174 @@ class NavigationBase extends React.Component {
+              <Link to="/namespaces">{linkerdWordLogo}</Link>
             </div>
             <IconButton className="drawer-toggle-btn" onClick={this.handleDrawerClick}>
-              {this.state.drawerOpen ? <ChevronLeftIcon /> : <MenuIcon />}
+              <ChevronLeftIcon />
             </IconButton>
           </div>
 
+          <Divider />
+          <MenuList>
+            <Typography variant="button" className={classes.sidebarHeading}>
+                Cluster
+            </Typography>
+
+            { this.menuItem("/namespaces", "Namespaces", namespaceIcon) }
+
+
+            { this.menuItem("/controlplane", "Control Plane",
+              <FontAwesomeIcon icon={faCloud} className={classes.shrinkCloudIcon} />) }
+
+          </MenuList>
+
           <Divider />
 
           <MenuList>
-            { this.menuItem("/overview", "Overview", <HomeIcon />) }
-            { this.menuItem("/tap", "Tap", <FontAwesomeIcon icon={faMicroscope} className={classes.shrinkIcon} />) }
-            { this.menuItem("/top", "Top", <FontAwesomeIcon icon={faStream} className={classes.shrinkIcon} />) }
-            { this.menuItem("/routes", "Top Routes", <FontAwesomeIcon icon={faRandom} className={classes.shrinkIcon} />) }
-            { this.menuItem("/servicemesh", "Service Mesh", <CloudQueueIcon className={classes.shrinkIcon} />) }
-            <NavigationResources />
+            <Button
+              variant="contained"
+              className={classes.namespaceChangeButton}
+              size="large"
+              onClick={this.handleNamespaceMenuClick}>
+              { formattedNamespaceName }
+              <ArrowDropDownIcon />
+            </Button>
+            <Menu
+              anchorEl={anchorEl}
+              open={this.state.namespaceMenuOpen}
+              keepMounted
+              onClose={this.handleNamespaceMenuClick}>
+              <MenuItem
+                value="all"
+                onClick={() => this.handleNamespaceChange("_all")}>
+                  All Namespaces
+              </MenuItem>
+
+              <Divider />
+
+              {namespaces.map(ns => (
+                <MenuItem
+                  onClick={() => this.handleNamespaceChange(ns.name)}
+                  key={ns.name}>
+                  {ns.name}
+                </MenuItem>
+              ))}
+            </Menu>
+            <Dialog
+              open={showNamespaceChangeDialog}
+              onClose={this.handleClose}
+              aria-labelledby="form-dialog-title">
+              <DialogTitle id="form-dialog-title">
+                Change namespace?
+              </DialogTitle>
+              <DialogContent>
+                <DialogContentText>
+              The resource you are viewing is in a different namespace than the namespace you have selected. Do you want to change the namespace from { selectedNamespace } to { newNamespace }?
```

Comment:
personally i didn't find this confirmation necessary. i was expecting that by changing the namespace while on a resource detail page, i would be taken to `/namespace/foo`. also it this was not expected, a user should be able to hit the back button.

Reply:
Thanks for the feedback. @grampelberg suggested the dialog in order to emulate the K8s dashboard. Once we get it in front of some users we can see what the preference is, I'm happy either way.